stackslib/src/net/api/getaccount.rs

```diff
@@ -0,0 +1,270 @@
+use crate::net::TipRequest;
+use crate::util_lib::boot::boot_code_id;
+use crate::util_lib::db::Error as DBError;
+
+use stacks_common::types::chainstate::StacksBlockId;
+use stacks_common::types::net::PeerHost;
+use stacks_common::util::hash::to_hex;
+use stacks_common::util::hash::Sha256Sum;
+
+use clarity::vm::clarity::ClarityConnection;
+use clarity::vm::costs::LimitedCostTracker;
+use clarity::vm::database::ClarityDatabase;
+use clarity::vm::database::STXBalance;
+use clarity::vm::representations::PRINCIPAL_DATA_REGEX_STRING;
+use clarity::vm::types::PrincipalData;
+use clarity::vm::types::StandardPrincipalData;
+use clarity::vm::ClarityVersion;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct AccountEntryResponse {
+    pub balance: String,
+    pub locked: String,
+    pub unlock_height: u64,
+    pub nonce: u64,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub balance_proof: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub nonce_proof: Option<String>,
+}
+
+#[derive(Clone)]
+pub struct RPCGetAccountRequestHandler {
+    pub account: Option<PrincipalData>,
+}
+impl RPCGetAccountRequestHandler {
+    pub fn new() -> Self {
+        Self { account: None }
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCGetAccountRequestHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(&format!(
+            "^/v2/accounts/(?P<principal>{})$",
+            *PRINCIPAL_DATA_REGEX_STRING
+        ))
+        .unwrap()
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body".to_string(),
+            ));
+        }
+
+        let account = if let Some(value) = captures.name("principal") {
+            PrincipalData::parse(value.into())
+                .map_err(|_e| Error::DecodeError("Failed to parse `principal` field".to_string()))?
+        } else {
+            return Err(Error::DecodeError(
+                "Missing in request path: `principal`".into(),
+            ));
+        };
+
+        self.account = Some(account);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+/// Handle the HTTP request
+impl RPCRequestHandler for RPCGetAccountRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.account = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
```

Comment:
These could also be references. `preamble` is only ever used as a reference it seems and I think `content` as well.